controllers/helmrepository_controller.go

```diff
@@ -209,11 +209,16 @@ func (r *HelmRepositoryReconciler) reconcile(ctx context.Context, repository sou
 		return sourcev1.HelmRepositoryNotReady(repository, sourcev1.IndexationFailedReason, err.Error()), err
 	}
 
-	// return early on unchanged generation
+	// return early on unchanged index
+	indexContent, err := yaml.Marshal(chartRepo.Index)
+	if err != nil {
+		return sourcev1.HelmRepositoryNotReady(repository, sourcev1.StorageOperationFailedReason, err.Error()), err
+	}
+	hash := r.Storage.Checksum(bytes.NewReader(indexContent))
 	artifact := r.Storage.NewArtifactFor(repository.Kind,
 		repository.ObjectMeta.GetObjectMeta(),
-		chartRepo.Index.Generated.Format(time.RFC3339Nano),
-		fmt.Sprintf("index-%s.yaml", url.PathEscape(chartRepo.Index.Generated.Format(time.RFC3339Nano))))
+		hash,
+		fmt.Sprintf("index-%s.yaml", url.PathEscape(hash)))
```

Comment:
```suggestion
		fmt.Sprintf("index-%s.yaml", hash))
```

As escaping is no longer required because the hash should be a normal string.